crates/net/eth-wire/src/types/broadcast.rs

```diff
@@ -687,30 +691,48 @@ pub struct RequestTxHashes {
     #[deref]
     #[deref_mut]
     #[into_iterator(owned, ref)]
-    hashes: Vec<TxHash>,
+    hashes: HashSet<TxHash>,
 }
 
 impl RequestTxHashes {
     /// Returns a new [`RequestTxHashes`] with given capacity for hashes. Caution! Make sure to
-    /// call [`Vec::shrink_to_fit`] on [`RequestTxHashes`] when full, especially where it will be
-    /// stored in its entirety like in the future waiting for a
+    /// call [`HashSet::shrink_to_fit`] on [`RequestTxHashes`] when full, especially where it will
+    /// be stored in its entirety like in the future waiting for a
     /// [`GetPooledTransactions`](crate::GetPooledTransactions) request to resolve.
     pub fn with_capacity(capacity: usize) -> Self {
-        Self::new(Vec::with_capacity(capacity))
+        Self::new(HashSet::with_capacity(capacity))
     }
-}
 
-impl FromIterator<(TxHash, Eth68TxMetadata)> for RequestTxHashes {
-    fn from_iter<I: IntoIterator<Item = (TxHash, Eth68TxMetadata)>>(iter: I) -> Self {
-        let mut hashes = Vec::with_capacity(32);
+    /// Returns an new empty instance.
+    fn empty() -> Self {
+        Self::new(HashSet::new())
+    }
 
-        for (hash, _) in iter {
-            hashes.push(hash);
+    /// Retains the given number of elements, returning and iterator over the rest.
+    pub fn retain_count(&mut self, count: usize) -> Self {
+        let rest_capacity = self.hashes.len().saturating_sub(count);
+        if rest_capacity == 0 {
+            return Self::empty()
         }
+        let mut rest = Self::with_capacity(rest_capacity);
 
-        hashes.shrink_to_fit();
+        let mut i = 0;
+        self.hashes.retain(|hash| {
+            i += 1;
+            if i <= count {
```

Comment:
can flip the order here to get rid of <= and skip +=1 if already > count

Reply:
count needs to increment every iteration

Reply:
ok you mean the true and false arms